### PR TITLE
test: E2E Search + Checkout

### LINKED ISSUE #12012: Request. E2E Search + Checkout
Please write an e2e test that combines search and checkout. 

There is already a test for search here: 
spartacus/projects/storefrontapp-e2e-cypress/cypress/integration/regression/product-search/product-search.e2e-spec.ts

There is also already a test for checkout here: 
spartacus/projects/storefrontapp-e2e-cypress/cypress/integration/regression/checkout/checkout-flow.e2e-spec.ts

However, there is no test that combines the 2 functionalities. 
**This is a basic business flow** that requires automated verification. 
There must be no doubt that product search and checkout work together from end-2-end. 
(SKUs must be added to cart and complete checkout after being listed as a result of a search)

This requested test may be long, but now, with parallelization this is no longer a concern.

### PR DESCRIPTION
**Pull request by bit172**
closes #12012

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  getSampleUser,
+  SampleCartProduct,
+  SampleProduct,
+  SampleUser,
+} from '../sample-data/checkout-flow';
+import * as checkout from './checkout-flow';
+import { AddressData, PaymentDetails } from './checkout-forms';
+import * as productSearchFlow from './product-search';
+
+export const firstAddToCartSelector = `${productSearchFlow.productItemSelector} cx-add-to-cart:first`;
```
Comment (dydome) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts]: not sure if it is required, I think we can put it directly in the cy.get

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+export function searchResultAddToCart() {
+  productSearchFlow.searchResult();
+}
+
+export function registerUser() {
```
Comment (dydome) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts]: Are you sure there is no ready method for this in another place?

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  return user;
+}
+
+export function addFirstResultToCartFromSearchAndLogin(sampleUser: SampleUser) {
+  cy.get(firstAddToCartSelector)
+    .findByText(/Add To Cart/i)
+    .click();
+  const loginPage = checkout.waitForPage('/login', 'getLoginPage');
+  cy.findByText(/proceed to checkout/i).click();
+  cy.wait(`@${loginPage}`);
```
Comment (dydome) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts]:
```suggestion
  cy.wait(`@${loginPage}`).its('response.statusCode').should('eq', 200);
```

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  checkout.loginUser(sampleUser);
```
Comment (dydome) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts]: shouldn't we log in the user first?

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+export function fillFormAndCheckout(
+  sampleUser: SampleUser,
+  cart: SampleCartProduct,
+  product: SampleProduct
+) {
+  checkout.fillAddressFormWithCheapProduct(sampleUser as AddressData, cart);
```
Comment (dydome) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts]: this method changed, please update

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  checkout.verifyDeliveryMethod();
+  checkout.fillPaymentFormWithCheapProduct(sampleUser as PaymentDetails);
```
Comment (dydome) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/product-search-checkout.ts]:
```suggestion
  checkout.fillPaymentFormWithCheapProduct(sampleUser);
```

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/product-search-checkout/product-search-checkout.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import { carts, products } from '../../../sample-data/product-search-checkout';
+
+context('Product Search and Checkout', () => {
+  viewportContext(['mobile', 'desktop'], () => {
+    beforeEach(() => {
+      cy.window().then((win) => win.sessionStorage.clear());
+    });
+
+    it('should search for a product, add it to cart and checkout', () => {
+      const productName = 'DSC-N1';
```
Comment (dydome) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/product-search-checkout/product-search-checkout.e2e-spec.ts]: shouldn't we use any product from helpers?

**Review by dydome — CHANGES_REQUESTED** [review]

**Comment by bit172** [comment]
These tests will be rewritten since most functions are outdated

**Review by dydome — APPROVED** [review]

**Pull request merged**